src/main/java/reposense/parser/CsvParser.java

```diff
@@ -43,27 +46,27 @@ public CsvParser(Path csvFilePath) throws IOException {
      */
     public List<T> parse() throws IOException {
         List<T> results = new ArrayList<>();
-        int lineNumber = 1;
 
         try (BufferedReader br = new BufferedReader(new FileReader(csvFilePath.toFile()))) {
             // Skip first line, which is the header row
             br.readLine();
             String line;
 
-            while ((line = br.readLine()) != null) {
+            for (int lineNumber = 2; (line = br.readLine()) != null; lineNumber++) {
                 String[] elements = line.split(ELEMENT_SEPARATOR);
 
                 if (line.isEmpty() || isLineMalformed(elements, lineNumber, line)) {
                     continue;
                 }
 
-                processLine(results, elements);
-                lineNumber++;
+                try {
+                    processLine(results, elements);
+                } catch (ParseException pe) {
+                    logger.warning(String.format(MESSAGE_LINE_PARSE_EXCEPTION_FORMAT, lineNumber, pe.getMessage()));
```

Comment:
Maybe we can add the line to the log message as well, similar to the malformed line log message.

Reply:
I realize my phrasing might be quite hard to understand. What I mean is this:

```
private static final String MESSAGE_LINE_PARSE_EXCEPTION_FORMAT =
    "Warning! Error parsing line %d in configuration file.\n"
    + "Contents: %s"        
    + "Error: %s";
```

```
logger.warning(String.format(MESSAGE_LINE_PARSE_EXCEPTION_FORMAT, lineNumber, line, pe.getMessage()));
```